asset_dashboard/static/js/components/maps/AssetsMap.js

```diff
@@ -0,0 +1,177 @@
+  const [searchText, setSearchText] = useState('')
+  const [searchedAssetType, setSearchedAssetType] = useState('buildings')
+
+  useEffect(() => {
+    if (props?.existing_geoms) {
+      setExistingGeoms(JSON.parse(props.existing_geoms))
+    }
+  }, [setSearchedGeoms, setExistingGeoms])
+
+  function onMapCreated(map) {
+    const group = new L.featureGroup()
+
+    map.eachLayer((layer) => {
+      if (layer.feature) {
+        group.addLayer(layer)
+      }
+    })
+
+    if (Object.keys(group._layers).length > 0) {
+      map.fitBounds(group.getBounds())
+    }
+  }
+
+  function onClipped(featureCollection) {
+    setClippedGeoms(featureCollection)
+  }
+
+  function saveGeometries() {
+    fetch(window.location.pathname, {
+        headers: {
+            'Accept': 'application/json',
+            'Content-Type': 'application/json',
+            'X-CSRFTOKEN': Cookies.get('csrftoken')
+        },
+        method: 'POST',
+        mode: 'same-origin',
+        body: JSON.stringify({
+          'geojson': clippedGeoms,
+          'phase': selectedPhase.value
+        })
+    }).then((response) => {
+      if (response.status == 200) {
+        location.reload()
+        // TODO: show success message
+        // https://stackoverflow.com/questions/13256817/django-how-to-show-messages-under-ajax-function
+      } else {
+        // TODO: catch error and show message
+      }
+    })
+  }
+
+  function searchAssets() {
+    const url = `/assets/?` + new URLSearchParams({
+      'q': searchText,
+      'asset_type': searchedAssetType
+    })
+
+    fetch(url, {
+      headers: {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json',
+        'X-CSRFTOKEN': Cookies.get('csrftoken')
+      },
+      mode: 'same-origin',
+      method: 'GET'
+    }).then((response) => response.json())
+    .then((data) => setSearchedGeoms(data))
+    .catch(error => {
+      // TODO: show an error message in the UI
+      console.log('error', error)
+    })
+  }
+ 
+  return (
+    <div className='row'>
+      <div className='col-4'>
+        <div className='row'>
+          <div className='input-group'>
+            <label htmlFor='asset-search' className='sr-only'>Search for Assets</label>
+            <input 
+              type='search'
+              onChange={(e) => setSearchText(e.target.value)}
+              value={searchText}
+              className='form-control' 
+              aria-label='Search for assets' 
+              placeholder='Search for assets' />
+            <select
+              className='form-control'
+              value={searchedAssetType}
+              onChange={(e) => setSearchedAssetType(e.target.value)}
+            >
+              <AssetTypeOptions />
+            </select>
+            <button 
+              onClick={() => searchAssets()}
+              className='btn btn-warning'>
+                Search
+            </button>
+          </div>
+        </div>
```

Comment:
IDK how long queries will be but this feels a bit cramped. What do you think about having this search take up the full width of the container (spanning both the results and map cols)?

<img width="1552" alt="Screen Shot 2022-01-20 at 11 28 54 AM" src="https://user-images.githubusercontent.com/12176173/150390833-a23b0244-e627-4547-a95b-bc2a639e8905.png">

Reply:
What about this? This is less cramped but doesn't span the entire box.

<img width="1155" alt="Screen Shot 2022-01-20 at 12 04 23 PM" src="https://user-images.githubusercontent.com/38969506/150396035-db1057e1-429d-4ef5-b2a6-f78aac15a3e3.png">


I wanted to keep the search elements separated from the right side of the page so that the search functionality is compartmentalized on the left side, and the add/edit function is on the right side. To me, that's a clearer user experience.

Reply:
thanks!